File: services/academic-service/src/main/java/com/visor/school/academicservice/repository/AcademicRecordRepository.java
package com.visor.school.academicservice.repository;

import com.visor.school.academicservice.model.AcademicRecord;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface AcademicRecordRepository extends JpaRepository<AcademicRecord, UUID> {
    Optional<AcademicRecord> findByStudentId(UUID studentId);
}

File: services/academic-service/src/main/kotlin/com/visor/school/academicservice/repository/AcademicRecordRepository.kt
package com.visor.school.academicservice.repository

import com.visor.school.academicservice.model.AcademicRecord
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional
import java.util.UUID

@Repository
interface AcademicRecordRepository : JpaRepository<AcademicRecord, UUID> {
    fun findByStudentId(studentId: UUID): Optional<AcademicRecord>
}

